docs/developer/README-cmake.md

```diff
@@ -35,7 +35,36 @@
 
 
 ### Arch Linux
-    pacman -S base-devel git cmake ninja nodejs clang boost libevent google-glog
+Install packages
+
+```
+pacman -S base-devel git cmake python2 ninja nodejs clang boost libevent google-glog
+```
+
+Set `python` to default to Python 2
+
+```
+ln -s /usr/bin/python2 $SOME_DIR/python
+export PATH=$SOME_DIR/python:$PATH
+```
+
+Disable warnings as errors by running `git apply` on the following patch
+
+```patch
+diff --git a/CMake/SkipCompiler.cmake b/CMake/SkipCompiler.cmake
+index c8ab080..f61d73c 100644
+--- a/CMake/SkipCompiler.cmake
++++ b/CMake/SkipCompiler.cmake
+@@ -52,7 +52,6 @@ if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR
+     $<$<CONFIG:Debug>:-g>
+     $<$<CONFIG:Release>:-DNDEBUG>
+     -Wall
+-    -Werror
```

Comment:
This is unfortunate. Should we just drop it altogether? cc @aorenste

Reply:
@Pavlos1 what is the error you are seeing?

Reply:
Unfortunate but reasonable given that it's only for a particular platform.  I wouldn't from Werror in general.  I doubt the issue is the gcc version - we were building regularly internally with gcc 5, 6 and 7.

Reply:
This is the specific error
```
third-party/install/include/folly/synchronization/AtomicStruct.h:77:11: error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of type ‘struct std::chrono::duration<long int, std::ratio<1, 1000000000> >’ with ‘private’ member ‘std::chrono::duration<long int, std::ratio<1, 1000000000> >::__r’ from an array of ‘using Raw = folly::_t<folly::detail::AtomicStructRaw<3> >’ {aka ‘long unsigned int’}; use assignment or copy-initialization instead [-Werror=class-memaccess]
     memcpy(&v, &d, sizeof(T));
     ~~~~~~^~~~~~~~~~~~~~~~~~~
```

Full cmd output (when re-run)

```
[1/138] Building CXX object lkg/runtime/native/CMakeFiles/sk_standalone.lkg.dir/src/sk_standalone.cpp.o
FAILED: lkg/runtime/native/CMakeFiles/sk_standalone.lkg.dir/src/sk_standalone.cpp.o 
/usr/bin/c++  -DUSE_JEMALLOC -Ithird-party/install/include -I../lkg/runtime/native/include -I../third-party/googletest/src/googletest/include -O2 -g -DNDEBUG -fPIC   -Wall -Werror -msse4.2 -Wno-sign-compare -std=c++14 -Wvla -fdata-sections -ffunction-sections -mcrc32 -DFOLLY_HAVE_MALLOC_H -O3 -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -Wmissing-declarations -Wno-vla -std=c++14 -MD -MT lkg/runtime/native/CMakeFiles/sk_standalone.lkg.dir/src/sk_standalone.cpp.o -MF lkg/runtime/native/CMakeFiles/sk_standalone.lkg.dir/src/sk_standalone.cpp.o.d -o lkg/runtime/native/CMakeFiles/sk_standalone.lkg.dir/src/sk_standalone.cpp.o -c ../lkg/runtime/native/src/sk_standalone.cpp
In file included from third-party/install/include/folly/detail/MemoryIdler.h:24,
                 from third-party/install/include/folly/synchronization/Baton.h:27,
                 from ../lkg/runtime/native/include/skip/Process.h:20,
                 from ../lkg/runtime/native/src/sk_standalone.cpp:29:
third-party/install/include/folly/synchronization/AtomicStruct.h: In instantiation of ‘static T folly::AtomicStruct<T, Atom>::decode(folly::AtomicStruct<T, Atom>::Raw) [with T = std::chrono::duration<long int, std::ratio<1, 1000000000> >; Atom = std::atomic; folly::AtomicStruct<T, Atom>::Raw = long unsigned int]’:
third-party/install/include/folly/synchronization/AtomicStruct.h:142:18:   required from ‘T folly::AtomicStruct<T, Atom>::load(std::memory_order) const [with T = std::chrono::duration<long int, std::ratio<1, 1000000000> >; Atom = std::atomic; std::memory_order = std::memory_order]’
third-party/install/include/folly/detail/MemoryIdler.h:144:34:   required from ‘bool folly::Baton<MayBlock, Atom>::tryWaitSlow(const std::chrono::time_point<Clock, Duration>&, const folly::WaitOptions&) [with Clock = std::chrono::_V2::steady_clock; Duration = std::chrono::duration<long int, std::ratio<1, 1000000000> >; bool MayBlock = true; Atom = std::atomic]’
third-party/install/include/folly/synchronization/Baton.h:176:5:   required from ‘void folly::Baton<MayBlock, Atom>::wait(const folly::WaitOptions&) [with bool MayBlock = true; Atom = std::atomic]’
third-party/install/include/folly/DefaultKeepAliveExecutor.h:42:33:   required from here
third-party/install/include/folly/synchronization/AtomicStruct.h:77:11: error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of type ‘struct std::chrono::duration<long int, std::ratio<1, 1000000000> >’ with ‘private’ member ‘std::chrono::duration<long int, std::ratio<1, 1000000000> >::__r’ from an array of ‘using Raw = folly::_t<folly::detail::AtomicStructRaw<3> >’ {aka ‘long unsigned int’}; use assignment or copy-initialization instead [-Werror=class-memaccess]
     memcpy(&v, &d, sizeof(T));
     ~~~~~~^~~~~~~~~~~~~~~~~~~
In file included from third-party/install/include/folly/detail/Futex.h:21,
                 from third-party/install/include/folly/MicroLock.h:24,
                 from ../lkg/runtime/native/include/skip/InternTable.h:17,
                 from ../lkg/runtime/native/include/skip/objects.h:14,
                 from ../lkg/runtime/native/include/skip/String.h:13,
                 from ../lkg/runtime/native/include/skip/Exception.h:11,
                 from ../lkg/runtime/native/src/sk_standalone.cpp:27:
/usr/include/c++/8.2.1/chrono:302:14: note: ‘struct std::chrono::duration<long int, std::ratio<1, 1000000000> >’ declared here
       struct duration
              ^~~~~~~~
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

Reply:
Also it looks like upstream `folly` has a fix(?) for it by casting the destination to a `void*`:
https://github.com/facebook/folly/blob/master/folly/synchronization/AtomicStruct.h#L77